eternalcore-core/src/main/java/com/eternalcode/core/feature/essentials/item/give/GiveService.java

```diff
@@ -0,0 +1,140 @@
+package com.eternalcode.core.feature.essentials.item.give;
+
+import com.eternalcode.core.configuration.implementation.PluginConfiguration;
+import com.eternalcode.core.injector.annotations.Inject;
+import com.eternalcode.core.injector.annotations.component.Service;
+import com.eternalcode.core.notice.NoticeService;
+import dev.triumphteam.gui.builder.item.ItemBuilder;
+import java.util.Optional;
+import org.bukkit.Material;
+import org.bukkit.enchantments.Enchantment;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+
+@Service
+public class GiveService {
+
+    private final int defaultGiveAmount;
+
+    private final PluginConfiguration pluginConfiguration;
+    private final NoticeService noticeService;
+
+    @Inject
+    public GiveService(PluginConfiguration pluginConfiguration, NoticeService noticeService) {
+        this.pluginConfiguration = pluginConfiguration;
+        this.noticeService = noticeService;
+
+        this.defaultGiveAmount = this.pluginConfiguration.items.defaultGiveAmount;
+    }
+
+    public void giveItem(Player player, Material material) {
+        if (this.isInvalidMaterial(material, player)) {
+            return;
+        }
+
+        this.giveItems(player, material, this.defaultGiveAmount, Optional.empty());
+    }
+
+    public void giveItem(Player player, Material material, int amount) {
+        if (this.isInvalidMaterial(material, player)) {
+            return;
+        }
+
+        this.giveItems(player, material, amount, Optional.empty());
+    }
+
+    public void giveItem(Player player, Material material, int amount, Enchantment enchantment, int level) {
+        if (this.isInvalidMaterial(material, player)) {
+            return;
+        }
+
+        this.giveItems(player, material, amount, Optional.of(new EnchantmentLevelPair(enchantment, level)));
+    }
+
+    private void giveItems(
+        Player player,
+        Material material,
+        int amount,
+        Optional<EnchantmentLevelPair> enchantmentLevel
+    ) {
+        int maxStackSize = material.getMaxStackSize();
+        int fullStacks = amount / maxStackSize;
+        int remainder = amount % maxStackSize;
+
+        for (int i = 0; i < fullStacks; i++) {
+            this.giveItemStack(player, material, maxStackSize, enchantmentLevel);
+        }
+
+        if (remainder > 0) {
+            this.giveItemStack(player, material, remainder, enchantmentLevel);
+        }
```

Comment:
This way to resolve items is not safe because you can give itemstack and next resolved itemstack could have not space to put inside player inventory.

Reply:
I think I fixed it, can you take a look?

https://github.com/user-attachments/assets/1cc70387-a66a-468a-9e5e-6b963b7ec03f

Reply:
This is not the case. You should switch config dropItems to false.